models/topics.model.js

```diff
@@ -18,14 +18,27 @@ exports.fetchArticleById = (id) => {
     });
 };
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `SELECT articles.author, articles.title,articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at ASC`
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+exports.fetchArticles = (query) => {
+  if (query) {
+    return db
+      .query(
+        `SELECT articles.author, articles.title,articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id WHERE articles.topic = $1 GROUP BY articles.article_id ORDER BY articles.created_at ASC`,
+        [query]
+      )
+      .then(({ rows }) => {
+        if (rows.length === 0) {
+          return Promise.reject({ status: 404, message: "not found" });
+        }
+        return rows;
+      });
+  } else
+    return db
+      .query(
+        `SELECT articles.author, articles.title,articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at ASC`
+      )
+      .then(({ rows }) => {
+        return rows;
+      });
```

Comment:
This approach works for the moment, but wouldn't if you were searching for articles with an existing topic but with no articles (which you'd expect an empty array for) 
You will need to check if the topic exists for this to work - make sure this is done using a query to the database 📝 
Can you please add this functionality and resubmit the ticket 🎫